### DISCUSSION TITLE
change graphQL request from GET to POST

### DISCUSSION THREAD

**Discussion by zielaq** [graphcommerce-org/graphcommerce#2351] [General]
How can i change graphql request from GET to POST and move all query to json body ? 

Sometimes i get 414 uri too long error with bigger query. 
Changing to post should resolve that issue,

**Comment by paales** [comment]
In the .meshrc.yaml there is a `useGETForQuery: true` configuration if you flip that to false it should use POST and run yarn codegen again it should switch to POST.

**Reply by zielaq** [reply]
@paales thanks !. but faced another issue. maxQueryComplexity should be 300 and my query has above 800. i'm using only default productPage2.graphql and its weird that this causes issues out of the box. 

as backend i'm using dockerized magento 2 image from bitnami

**Comment by paales** [comment]
So there is a bug in one of the libraries used, which has since been fixed: https://github.com/ardatan/graphql-tools/releases/tag/release-1723045008485

To install that version, this should probably work:

```
yarn up -R "@graphql-mesh/*" "@graphql-tools/*"
```

**Reply by zielaq** [reply]
@paales thanks, updating package.json deps helped. in that case i was able to switch back to GET request instead of POST. 
but sometimes i'm still getting Max query complexity error. F.eg. adding something with variants to cart shows that complexity should be 300 but got 356
